common/workunit/workunit.cpp

```diff
@@ -10149,6 +10149,23 @@ extern WORKUNIT_API bool secDebugWorkunit(const char * wuid, ISecManager &secmgr
     return false;
 }
 
+bool isResultASet(IWUResult *r)
+{
+    SCMStringBuffer schema;
+    r->getResultEclSchema(schema);
+    return !strnicmp(schema.str(), "RECORD set of", strlen("RECORD set of"));
```

Comment:
In my testing, even for a scalar value the stored ECL schema is a complete record definition, something like:

RECORD string value{NAMED('str')}; END;

For a set it looks like:

RECORD set of string value{NAMED('st')}; END;